libfreerdp/core/connection.c

```diff
@@ -314,8 +314,18 @@ BOOL rdp_client_disconnect(rdpRdp* rdp)
 BOOL rdp_client_redirect(rdpRdp* rdp)
 {
 	BOOL status;
+	BOOL reconnect_channels = FALSE;
 	rdpSettings* settings = rdp->settings;
 
+	/* Ensure existing channels are disconnected */
+	rdpContext* context = rdp->context;
+	rdpChannels* channels = context->channels;
+	if (channels->connected)
```

Comment:
@bjcollins Good point. Either way it may be a good idea to also have a look at ```rdp_client_reconnect``` and keep it synced with this function.